lib/hanami/action/flash.rb

```diff
@@ -55,28 +52,26 @@ def []=(key, value)
       # @since 0.3.0
       # @api private
       def [](key)
-        @merged.fetch(key) do
-          _values.find {|data| !data[key].nil? }
-        end
+        _data[key] || search_in_kept_data(key)
```

Comment:
This should still use `fetch`.

What we're looking for here is the presence of `key` in the internal data. `fetch` yields the block if the key is missing, which is the wanted behavior.

---

The syntax with `||` on the other hand, yields the rightmost expression both if the key is missing (good), but also if the key is present but the value is `nil` (bad).

Reply:
Great point @jodosha I will use `fetch`